src/app/clusters/device-energy-management-server/device-energy-management-server.cpp

```diff
@@ -104,49 +104,55 @@ CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValu
 }
 
 // CommandHandlerInterface
-CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context)
+CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & cluster,
+                                               DataModel::ListBuilder<DataModel::AcceptedCommandEntry> & builder)
 {
     using namespace Commands;
+    using QF                            = DataModel::CommandQualityFlags;
+    static const auto kDefaultFlags     = chip::BitFlags<QF>(QF::kTimed, QF::kLargeMessage, QF::kFabricScoped);
+    static const auto kDefaultPrivilege = chip::Access::Privilege::kOperate;
 
     if (HasFeature(Feature::kPowerAdjustment))
     {
-        for (auto && cmd : {
-                 PowerAdjustRequest::Id,
-                 CancelPowerAdjustRequest::Id,
-             })
-        {
-            VerifyOrExit(callback(cmd, context) == Loop::Continue, /**/);
-        }
+        ReturnErrorOnFailure(builder.AppendElements({
+            { PowerAdjustRequest::Id, kDefaultFlags,
+              kDefaultPrivilege }, // TODO:: WHAT IS THE NEEDED PRIVILEGE AND CommandQualityFlags
```

Comment:
Flags are from the spec.
In this case the privilege seems to be operate and no special flags:

![image](https://github.com/user-attachments/assets/71d31d4e-5a75-4f85-8b07-5e270c392a41)

XML would contain special flags: https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap-templates/zcl/data-model/chip/device-energy-management-cluster.xml#L118 however commands default to operate.


The code generation from ember is in `CommandNeedsTimedInvoke`, `CommandIsFabricScoped` and `CommandHasLargePayload`. See https://github.com/project-chip/connectedhomeip/blob/master/zzz_generated/app-common/app-common/zap-generated/cluster-objects.cpp#L36615 for generated code.